### PR TITLE
refactor: Converts AssessmentItemMastery into an EventSourcedEntity

### PR DESCRIPTION
**Pull request by milana-todorovic**

### CODE REVIEW

File: src/Tutor.Core/LearnerModel/DomainOverlay/KnowledgeComponentMasteries/AssessmentItemMastery.cs [hunk 1]
```diff
@@ -1,25 +1,64 @@
...
+            {
+                AssessmentItemId = AssessmentItemId,
+                Submission = submission,
+                Evaluation = evaluation
+            });
+        }
+
+        public Result SeekHints()
+        {
+            // TODO: validation logic
```
Comment (Luburic) [inline src/Tutor.Core/LearnerModel/DomainOverlay/KnowledgeComponentMasteries/AssessmentItemMastery.cs:40]: Remove the TODOs

**Review by Luburic — APPROVED** [review]
Looks good. You can remove the comments and merge unless there is something else that needs to be wrapped up.

File: src/Tutor.Core/LearnerModel/DomainOverlay/KnowledgeComponentMasteries/AssessmentItemMastery.cs (hunk 1 above)
Comment (milana-todorovic) [reply src/Tutor.Core/LearnerModel/DomainOverlay/KnowledgeComponentMasteries/AssessmentItemMastery.cs]: Removed in dda0d5278a8b594d2a8025b1344dc124ca7ddbb2.

**Pull request merged**